chain/chain/src/garbage_collection.rs

```diff
@@ -173,6 +181,14 @@ impl Chain {
                     break;
                 } else if prev_block_refcount == 1 {
                     debug_assert_eq!(blocks_current_height.len(), 1);
+                    if let Some(ref validator_account_id) = validator_account_id {
+                        chain_store_update.clear_obsolete_shard_data(
```

Comment:
How does doing this here (when prev_block_refcount == 1) guarantee that for any shard S, the memtrie of S is unloaded before the next time we start state sync for S (which... could be as soon as in the same epoch? I'm not really sure).

On top of that, should we add an assert when state sync starts that we do not have memtrie for that shard?